site-src/geps/gep-1016.md

```diff
@@ -0,0 +1,781 @@
+# GEP-1016: GRPCRoute
+
+* Issue: [#1016](https://github.com/kubernetes-sigs/gateway-api/issues/1016)
+* Status: Implementable
+
+(See definitions in [Kubernetes KEP][kep-status].
+
+[kep-status]: https://github.com/kubernetes/enhancements/blob/master/keps/NNNN-kep-template/kep.yaml#L9
+
+## Goal
+
+Add an idiomatic GRPCRoute for routing gRPC traffic.
+
+## Non-Goals
+
+While certain gRPC implementations support multiple transports and multiple
+interface definition languages (IDLs), this proposal limits itself to
+[HTTP/2](https://developers.google.com/web/fundamentals/performance/http2) as
+the transport and [Protocol Buffers](https://developers.google.com/protocol-buffers)
+as the IDL, which makes up the vast majority of gRPC traffic in the wild.
+
+## Introduction
+
+While it would be possible to support gRPC via CRDs, in the long run, this would
```

Comment:
To be clear, the whole API is done with CRDs. Do you mean custom, out-of-tree resources?

Reply:
Yes. Clarified.